pgjdbc/src/main/java/org/postgresql/util/PGrange.java

```diff
@@ -0,0 +1,360 @@
+   *                       {@code false} to make the lower bound is exclusive
+   */
+  public void setLowerInclusive(boolean lowerInclusive) {
+    this.lowerInclusive = lowerInclusive;
+  }
+
+  /**
+   * Checks if the upper bound is inclusive.
+   *
+   * @return {@code true} if the upper bound is inclusive, {@code false} if the upper bound is exclusive
+   */
+  public boolean isUpperInclusive() {
+    return this.upperInclusive;
+  }
+
+  /**
+   * Sets the upper bound inclusive.
+   *
+   * @param upperInclusive {@code true} to make the upper bound is inclusive,
+   *                       {@code false} to make the upper bound is exclusive
+   */
+  public void setUpperInclusive(boolean upperInclusive) {
+    this.upperInclusive = upperInclusive;
+  }
+
+  /**
+   * Return the lower bound.
+   *
+   * @return the lower bound, {@code null} if the lower bound is infinite
+   * @see #isLowerInfinite()
+   */
+  public @Nullable T getLowerBound() {
+    return this.lowerBound;
+  }
+
+  /**
+   * Sets the lower bound.
+   *
+   * @param lowerBound the lower bound to set,
+   *                   {@code null} to set the lower bound to infinite
+   */
+  public void setLowerBound(@Nullable T lowerBound) {
+    this.lowerBound = lowerBound;
+  }
+
+  /**
+   * Return the upper bound.
+   *
+   * @return the upper bound, {@code null} if the upper bound is infinite
+   * @see #isUpperInfinite()
+   */
+  public @Nullable T getUpperBound() {
+    return this.upperBound;
+  }
+
+  /**
+   * Sets the upper bound.
+   *
+   * @param upperBound the upper bound to set,
+   *                   {@code null} to set the upper bound to infinite
+   */
+  public void setUpperBound(@Nullable T upperBound) {
+    this.upperBound = upperBound;
+  }
+
+  /**
+   * Checks if the lower bound is infinite.
+   *
+   * @return {@code true} if the lower bound is infinite, {@code false} if the lower bound is set
+   */
+  public boolean isLowerInfinite() {
+    return this.lowerBound == null;
+  }
+
+  /**
+   * Checks if the upper bound is infinite.
+   *
+   * @return {@code true} if the upper bound is infinite, {@code false} if the upper bound is set
+   */
+  public boolean isUpperInfinite() {
+    return this.upperBound == null;
+  }
+
+  @Override
+  public String getValue() {
+    return this.isEmpty() ? "empty" : (this.lowerInclusive ? '[' : '(')
+        + (this.lowerBound == null ? "" : this.serializeBound(this.lowerBound))
+        + ','
+        + (this.upperBound == null ? "" : this.serializeBound(this.upperBound))
+        + (this.upperInclusive ? ']' : ')');
+  }
+
+  @Override
+  public void setValue(@Nullable String value) throws SQLException {
+    super.setValue(value);
+    if ("empty".equals(value) || value == null) {
+      this.lowerInclusive = false;
+      this.upperInclusive = false;
+      this.lowerBound = null;
+      this.upperBound = null;
+      this.empty = true;
```

Comment:
If value is null, then shouldn't this.isNull() be true? Since this sets this.empty = true, getValue will not return null, an consequently isNull() while return false. That does not seem correct.